tasks/new_e2e_tests.py

```diff
@@ -148,44 +148,40 @@ def _clean_locks():
     for entry in os.listdir(Path(lock_dir)):
         subdir = os.path.join(lock_dir, entry)
         for filename in os.listdir(Path(subdir)):
-            file_path = os.path.join(subdir, filename)
-            if os.path.isfile(file_path) and filename.endswith(".json"):
-                os.remove(file_path)
-                print(f"🗑️ Deleted lock: {file_path}")
+            path = os.path.join(subdir, filename)
+            if os.path.isfile(path) and filename.endswith(".json"):
+                os.remove(path)
+                print(f"🗑️ Deleted lock: {path}")
+            elif os.path.isdir(path):
```

Comment:
Can you explain why you added this case?

Reply:
Pulumi recently changed the way the name stack adding organisation and project, this has impact also on how locks are stored. Before this change, we were looking at `.json` file on root `locks` folder as they were all there. Now locks are in `.locks/<organisation>/<project>/<stack>.json`. I could change this to recursively look for `json` file and remove only them, but it seems to be safe to remove recursively all subfolders directly.